File: .NetServer/Vikreta/Entities/Bid.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Vikreta.Entities
{
    public class Bid : BaseEntity
    {
        public long ProductId { get; set; }
        public long BuyerId { get; set; }
        [Range(0.01, 999999999999999999.99, ErrorMessage = "Bid amount must be between 0.01 and 999999.99.")]
        public decimal BidAmount { get; set; }


        [Required]
        [ForeignKey("ProductId")]
        public Product Product { get; set; }

        [Required]
        [ForeignKey("BuyerId")]
        public User Buyer { get; set; }
    }
}

File: JavaServer/src/main/java/com/vikreta/entities/Bid.java
package com.vikreta.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;


@Getter
@Setter
@Entity
@NoArgsConstructor
@Table(name = "bids")
public class Bid extends BaseEntity {
	
	@Column
	private double bidAmount;

	@ManyToOne
	@JoinColumn(name = "product_id",nullable = false)
	private Product product;
	
	@ManyToOne
	@JoinColumn(name = "buyer_id",nullable = false )
	private User buyer;

}
